examples/plot_mne_epoch_masking.py

```diff
@@ -41,6 +41,10 @@
 # results.
 
 # create the input array for Comodulogram.fit
+
+# low_sig, high_sig = raw[8, :][0], raw[10, :][0]
+# mask = MaskIterator(events, tmin, tmax, raw.n_times, raw.info['sfreq'])
+
 low_sig, high_sig, mask = raw_to_mask(raw, ixs=ixs, events=events, tmin=tmin,
                                       tmax=tmax)
```

Comment:
I couldn't quite understand this example. The `mask` object that I get is an iterable? That means when you provide it to `estimator.fit` in L55, it computes it for all the iterations or just the first one?

Reply:
Which iterations are you talking about?
The `mask` object is an iterable, that creates on the fly a series of `n_events * n_tmin_tmax` arrays.
Each array is used only once, after filtering, and before computing the PAC modulation index, to select some time windows around events.
In the end, you get `n_events * n_tmin_tmax` comodulograms.

Reply:
So, in this particular example, if I do:

```py
>>> events.shape
(319, 3)
```

but then,

```py
>>> estimator.comod_.shape
(6, 20, 80)
```

so what is the shape 6 x 20 x 80. I guess 20 is `low_freq_range.shape` and 80 is `high_freq_range.shape` (aside, might be nice to recover the `high_freq_range` array when it's "auto"). But what is 6?

Reply:
6 is the number of event kinds: `len(np.unique(events[:, 2]))`

Reply:
I see, now I understand what you have done in the code. I had missed the fact that you filter the data only once ...

The use-case is the following: I have bad segments in my data that I would not like to use for the PAC estimation. One way to deal with it is to compute the PAC on long overlapping windows (excluding the bad segments) and then get some sort of summary statistic of the comodulogram over these windows.  I thought I might be able to create masks for these windows and get `n_mask` comodulograms. While what you do is neat for preventing edge artifacts, it wouldn't work in my case as the filtering would spread the artifacts ...

I suppose I'll resort to doing my own loop. Thanks for explaining.

Reply:
> Which iterations are you talking about?

sorry, it was my bad. I was thinking that `len(masks)` was `n_events`